base/path.jl

```diff
@@ -63,17 +63,17 @@ Return the current user's home directory.
 """
 function homedir()
     path_max = 1024
-    buf = Vector{UInt8}(undef, path_max)
+    buf = Base.StringVector(path_max + 1)
     sz = RefValue{Csize_t}(path_max + 1)
```

Comment:
```suggestion
    sz = RefValue{Csize_t}(length(buf))
```

Reply:
Should just be `length(buf)` with no `+1`, no?

Reply:
Oh I see, see below.